### PR TITLE
[ioctl] Refactor bcinfo command in new ioctl

### LINKED ISSUE #3268: [ioctl] refactor bcinfo command in new ioctl
### What would you like to be added
 - Step 1: refactor  command  [newcmd/bc/bcinfo.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/bc/bcinfo.go) 
 - Step 2: append necessary unit test case in [newcmd/bc/bcinfo_test.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/bc/bcinfo_test.go) 

#### Notes:
1. Use [function of client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) . 
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/bc/bcinfo.go ).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `bcinfo` command in new `ioctl`, with the following note.

* Update all function to [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go)
Replace `fmt.Println` with `cmd.Println`
[Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. e.g. `output.PrintError` could be replaced with `errors.Wrap`.
Global variables (`config.ReadConfig` `config.DefaultConfigFile` etc.) have been wrapped into [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go).
Refactor unit test to cover the modification.

Fixes #3268

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] TestNewBCInfoCmd

**Test Configuration**:

- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/newcmd/bc/bcinfo.go [hunk 1]
```diff
@@ -8,7 +8,9 @@ package bc
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"log"
 
 	"github.com/spf13/cobra"
```
Comment (huof6829) [inline ioctl/newcmd/bc/bcinfo.go:16]: delete space line

File: ioctl/newcmd/bc/bcinfo.go [hunk 2]
```diff
@@ -67,16 +62,21 @@ func NewBCInfoCmd(client ioctl.Client) *cobra.Command {
 			}
```
Comment (huof6829) [inline ioctl/newcmd/bc/bcinfo.go:62]: can use GetChainMeta() in bc.go  for line51-62 ?

File: ioctl/newcmd/bc/bc.go [hunk 3]
```diff
@@ -87,9 +88,9 @@ func GetChainMeta(client ioctl.Client) (*iotextypes.ChainMeta, error) {
 	if err != nil {
 		sta, ok := status.FromError(err)
 		if ok {
-			return nil, output.NewError(output.APIError, sta.Message(), nil)
+			return nil, errors.Wrap(nil, sta.Message())
 		}
-		return nil, output.NewError(output.NetworkError, "failed to invoke GetChainMeta api", err)
+		return nil, errors.Wrap(err, "failed to invoke GetChainMeta api")
```
Comment (huof6829) [inline ioctl/newcmd/bc/bc.go:93]: pls modify output.NewError in this file. thx

File: ioctl/newcmd/bc/bc.go [hunk 4]
```diff
@@ -87,9 +88,9 @@ func GetChainMeta(client ioctl.Client) (*iotextypes.ChainMeta, error) {
 	if err != nil {
```
Comment (huof6829) [inline ioctl/newcmd/bc/bc.go:88]: add https://github.com/iotexproject/iotex-core/blob/8ec14b5c5bb76c31660184130361318ea38e4659/ioctl/cmd/bc/bc.go#L75-L78

File: ioctl/newcmd/bc/bcinfo.go [hunk 5]
```diff
@@ -53,30 +44,27 @@ func NewBCInfoCmd(client ioctl.Client) *cobra.Command {
...
+			cmd.Println(message.String())
 			return nil
 		},
 	}
 
-	flagEndpointUsage, _ := client.SelectTranslation(_flagEndpointUsages)
-	flagInsecureUsage, _ := client.SelectTranslation(_flagInsecureUsages)
-
 	cmd.PersistentFlags().StringVar(&endpoint, "endpoint", client.Config().Endpoint, flagEndpointUsage)
 	cmd.PersistentFlags().BoolVar(&insecure, "insecure", !client.Config().SecureConnect, flagInsecureUsage)
```
Comment (huof6829) [inline ioctl/newcmd/bc/bcinfo.go:59]: delete , because flag is set in NewBCCmd

**Review by huof6829 — APPROVED** [review]

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**